### PR TITLE
Add ability to export to CSV, import from CSV

### PR DESCRIPTION
**Pull request by leeyiheng12**
Add buttons in the dropdown menu bar to export to CSV, and import from CSV.

Add new classes to encapsulate the storing of the address book in CSV format, as well as writing data to CSV format and parsing from CSV format safely, and new classes to encapsulate a Person class and Tag class in CSV format.

Add documentation and unit tests to new classes created.

### CODE REVIEW

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -78,4 +80,36 @@ public GuiSettings getGuiSettings() {
...
+    @Override
+    public boolean loadAddressBookFromCsv(Path csvFilePath) {
+        boolean result;
+        try {
+            Optional<ReadOnlyAddressBook> ab = storage.readAddressBookFromCsvFile(csvFilePath);
+            if (ab.isPresent()) {
+                model.setAddressBook(ab.get());
+                result = true;
+            } else {
+                System.out.println("Error");
```
Comment (jetrz) [inline src/main/java/seedu/address/logic/LogicManager.java]: Is this line needed? (Line 106)

File: src/main/java/seedu/address/storage/CsvAdaptedPerson.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Address.class.getSimpleName()));
+        }
+        if (!Address.isValidAddress(address)) {
+            throw new IllegalValueException(Address.MESSAGE_CONSTRAINTS);
+        }
+        final Address modelAddress = new Address(address);
+
+        final Set<Tag> modelTags = new HashSet<>(personTags);
+        return new Person(modelName, modelPhone, modelEmail, modelInsurancePackage, modelAddress, modelTags);
+    }
```
Comment (jetrz) [inline src/main/java/seedu/address/storage/CsvAdaptedPerson.java:148]: Very neat and simple to understand!

File: src/main/resources/view/MessageWindow.fxml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        <opaqueInsets>
+          <Insets bottom="10.0" left="5.0" right="10.0" top="5.0" />
+        </opaqueInsets>
+        <padding>
+          <Insets bottom="10.0" left="5.0" right="10.0" top="5.0" />
+        </padding>
+      </HBox>
+    </Scene>
+  </scene>
+</fx:root>
```
Comment (jetrz) [inline src/main/resources/view/MessageWindow.fxml:38]: Nice job on editing front-end files so early on!

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 4]
```diff
@@ -147,6 +156,102 @@ public void handleHelp() {
...
+        chooser.setSelectedFile(new File("ClientConnectData.csv"));
+
+        // opens the dialog
+        int returnVal = chooser.showSaveDialog(null);
+        Path chosenDirectoryPath = null;
+        if (returnVal == JFileChooser.APPROVE_OPTION) {
+            chosenDirectoryPath = chooser.getSelectedFile().toPath().toAbsolutePath();
+        }
+        return chosenDirectoryPath;
+    }
```
Comment (jetrz) [inline src/main/java/seedu/address/ui/MainWindow.java:264]: Could these error messages be stored in external variables to reduce the duplication of code and make it more understandable?

File: src/main/java/seedu/address/commons/util/CsvUtil.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+        try {
+            s = new Scanner(new File(filePath.toString())); // create a Scanner using the File as the source
+        } catch (IOException err) {
+            System.out.println("Error");
+            return persons;
+        }
+
+        s.nextLine();  // headers
+        System.out.println("hello");
```
Comment (jetrz) [inline src/main/java/seedu/address/commons/util/CsvUtil.java]: Is this line required? (Line 48)

**Review by jetrz — COMMENTED** [review]
Overall very well done! This feature was not easy to implement, as seen from the no. of LoC changed. Only have minor comments for changes in abstraction for error messages, and some possibly unnecessary code. 

Thanks for the effort Yi Heng!

File: src/main/java/seedu/address/logic/LogicManager.java (hunk 1 above)
Comment (leeyiheng12) [reply src/main/java/seedu/address/logic/LogicManager.java]: Oh I did this to remind myself to add changes, but must have forgotten about this afterwards, will change this next time!

File: src/main/java/seedu/address/commons/util/CsvUtil.java (hunk 5 above)
Comment (leeyiheng12) [reply src/main/java/seedu/address/commons/util/CsvUtil.java]: Forgot about this too oops...

File: src/main/java/seedu/address/logic/LogicManager.java (hunk 1 above)
Comment (leeyiheng12) [reply src/main/java/seedu/address/logic/LogicManager.java]: Wait idk what happened but I ended up changing this code in the end but it's not reflected here

**Pull request merged**